cmd/mobile-security-service/main.go

```diff
@@ -56,12 +57,23 @@ func initLogger(level, format string) {
 
 // Invoke handlers, services and repositories here
 func initHandlers(e *echo.Echo, c config.Config) {
+
 	// Prefix api routes
 	apiRoutePrefix := c.ApiRoutePrefix
 	apiGroup := e.Group(apiRoutePrefix)
 
```

Comment:
The handler lawyer is responsible to receive the requests. The lawyer which will connect to the database is the repository one. So, I am not sure about this code here. I mean I don't think it should be imported in this lawyer at all.

Reply:
@camilamacedo86 what is a lawyer? I'm not familiar with this term in this context.

Perhaps this function should not be called `initHandlers` as you are right, it is doing more than this, it is initialising the service and repository as well.

Reply:
I would be inclined to split the db connect to it's own function, called from main before echo is instansiated and pass it through to this function.

Reply:
@craicoverflow I think @camilamacedo86 means `layer`

+1 on `initHandlers()` not being the best place for db setup, perhaps move the db setup either to `main()` or its own dedicated method that would be called in `main()`, or else rename `initHandlers` to reflect what it's now doing?

Reply:
Hi @craicoverflow,

`Layer` it is regarding the architecture of the project. Each layer has one kind of responsibility. 

So, the request will go through them as follows. 

https://cdn-images-1.medium.com/max/1600/1*0bgo3z8DsVR0WbEA-D6FEg.png

In this way, the layer responsible to connect to the database and get the data "Data Access Object" here is the repository

Reply:
> Hi @craicoverflow,
> 
> `Layer` it is regarding the architecture of the project. Each layer has one kind of responsibility. So, the request will go through them as follows.
> 
> ![1_0bgo3z8dsvr0wbea-d6feg](https://user-images.githubusercontent.com/7708031/52492072-63d30600-2bc0-11e9-92f7-bc5db8f52360.png)
> 
> In this way, the layer responsible to connect to the database and get the data "Data Access Object" here is the repository. The handlers should not call the db directly.

Oh _layer_! You had said _lawyer_ which really confused me :grin:

Reply:
The Grammar app sometimes leads me to some misunderstands. Sorry, for that.